### PR TITLE
enable editing sequence chapters

### PR DESCRIPTION
**Pull request by s-cheng**
This PR exposes the existing chapter editing UI to the proper users.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204254699363093) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/sequences/ChaptersItem.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const ChaptersItem = ({ chapter, canEdit, classes }: {
           })}
         </AnalyticsContext>
       </div>
-      {!chapter.title && canEdit && <SectionFooter>{editButton}</SectionFooter>}
+      {canEdit && <SectionFooter>{editButton}</SectionFooter>}
```
Comment (s-cheng) [inline packages/lesswrong/components/sequences/ChaptersItem.tsx:76]: I don't understand why we had this extra condition here - maybe @jimrandomh or @Raemon knows? I was asked to edit a chapter description and I just made this change locally and was able to edit it, so it seems fine to keep?

**Comment by jpaddison3** [comment]
Given that basically this entire review is figuring out why Chesterton put that fence there, I'm gonna wait on a reply from LW

**Comment by Raemon** [comment]
Reasons for this fence:

1. In general we hadn't really optimized the chapter UI to be comprehensible. I think it's at least somewhat confusing if you've never made a sequence before. You might want to do some user testing to find out how confusing it is and whether there are ways to somewhat clarify it.

2. There'll be temptation will be for users to make giant complicated sequences with lots of chapters. This... maybe is the best thing users can do given the current tools available to them, but it's not obviously better than creating multiple sequences. (In particular, when you hover over a sequence and see the post-list, really long lists of posts are very overwhelming. The Replacing Guilt sequence should really be broken into multiple sequences in a higher-order collection. CFAR handbook is another contender for "maybe too long.")

But I guess we don't give users the ability to make books or collections, so maybe what needs to happen is to improve the UI on hoverover. Maybe make it only expand one chapter by default. 

Replacing Guilt
https://www.lesswrong.com/s/pFatcKW3JJhTSxqAF

CFAR Handbook
https://www.lesswrong.com/sequences/KAv8z6oJCTxjR8vdR

**Comment by Raemon** [comment]
I don't think it's crazy to expose it to other users without any reflection, but I think it warrants a bit more improvement if you're gonna do it.

**Comment by Raemon** [comment]
Oh actually, I wrote this reply before reading what this PR actually did. I'm... less confident why this particular condition is here.

File: packages/lesswrong/components/sequences/SequencesPage.tsx
Comment (s-cheng) [inline packages/lesswrong/components/sequences/SequencesPage.tsx:1]: @Raemon Luckily(?) I ended up needing to edit chapters again, and I think I've figured out what's going on. I just want to be able to edit chapters as an admin, not trying to open the door for users. I changed `canEdit` for `ChaptersList` to check the proper permissions.

Previously it was checking the **sequence's** permissions, so probably was causing the "edit chapter" button to appear for the user who owned the sequence, which I assume was mitigated adding the check for `!chapter.title`. Creating a chapter is separate from editing a chapter, and that is already checking the proper permissions.

@jpaddison3 I think this is ready for review now

**Pull request merged**